src/plugins/badge.coffee

```diff
@@ -608,4 +610,45 @@ class Badge
 
     "#{settings.url.application_base}/badge/shop_reviews?#{serialized_params}"
 
+  _initHideOnScroll: (badge) =>
+    past_scroll = window.parent.pageYOffset
+    return unless past_scroll? # pageYOffset is not supported for IE < 9
+
+    @_attachEvent @parent_doc, 'scroll', (_event) =>
+      current_scroll = window.parent.pageYOffset
+
+      return if current_scroll == past_scroll # Ignore horizontal scrolling
+      return if @_outOfBound(current_scroll) # Ignore OSX bounce
+
+      scroll_diff = current_scroll - past_scroll
+      return if Math.abs(scroll_diff) <= 5
+
+      badge.style.display = if scroll_diff > 0 then 'none' else 'block'
+
+      past_scroll = current_scroll
+
+  _outOfBound: (current_scroll) ->
```

Comment:
Alternative that uses early return, without making all the calculations for the bottom bounce:

```
  _outOfBound: (current_scroll) ->
    return true if current_scroll < 0 #top bounce

    current_scroll + @_vieport().height > @_document_height() #bottom bounce

  _document_height: ->
    Math.max @parent_doc.body.scrollHeight, @parent_doc.documentElement.scrollHeight,
             @parent_doc.body.offsetHeight, @parent_doc.documentElement.offsetHeight,
             @parent_doc.body.clientHeight, @parent_doc.documentElement.clientHeight
```